### DISCUSSION TITLE
GraphQL queries from within built pages

### DISCUSSION THREAD

**Discussion by 3goats** [squidfunk/mkdocs-material#3768] [General]
Hi I just signed up to the insiders repo.

Wondering if there's a way to invoke a GraphQL query from with a doc page and have it rendered into a widget or code fence within the doc. I need this to be a dynamic capability from within the generated pages as oppose to the build process.

**Comment by squidfunk** [comment]
You can add whatever additional logic you like as part of [additional JavaScript](https://squidfunk.github.io/mkdocs-material/customization/#additional-javascript). The theme has [many parts that can be hooked into](https://github.com/squidfunk/mkdocs-material/discussions/3429#discussioncomment-1910541).

**Reply by 3goats** [reply]
Perfect thanks. Assume I would be able to link this up with a button component?

**Reply by squidfunk** [reply]
Yes, the button classes just add styling.
